### PR TITLE
Update apps

### PR DESCRIPTION
**Pull request by spalmer25**

### CODE REVIEW

File: src/globals.h [hunk 1]
```diff
@@ -37,23 +37,11 @@ extern uint8_t G_io_seproxyhal_spi_buffer[IO_SEPROXYHAL_BUFFER_SIZE_B];
 #define MAX_PARTNER_NAME_LENGHT 15
 
 #define PARTNER_NAME_PREFIX_FOR_FUND "To "
-#define PARTNER_NAME_PREFIX_SIZE     (sizeof(PARTNER_NAME_PREFIX_FOR_FUND) - 1)  // Remove trailing '\0'
 
-#pragma pack(push, 1)
 typedef struct partner_data_s {
-    uint8_t name_length;
```
Comment (ajinkyasraj) [inline src/globals.h:44]: Do you think this change should be part of this PR? Also,  this seems to affect the original app-exchange and not related to tezos. Better to suggest these improvements separately and not as part of tezos upgrade.

File: src/set_partner_key.c [hunk 2]
```diff
@@ -73,11 +73,14 @@ int set_partner_key(const command_t *cmd) {
         return reply_error(err);
     }
 
-    // The incoming partner name is NOT NULL terminated, so we use memcpy and
-    // manually NULL terminate the buffer
-    memset(G_swap_ctx.partner.name, 0, sizeof(G_swap_ctx.partner.name));
```
Comment (ajinkyasraj) [inline src/set_partner_key.c:78]: name and prefix are part of union. Any perticular reason you have combined to just one char array? I dont see any benefit in terms of size reduction. Union is better in terms of making it clear for what we are using the array?

File: src/globals.h (hunk 1 above)
Comment (spalmer25) [reply src/globals.h:44]: The first commit is the update of the `app-exchange`.
It's a merge between our develop branch and theirs.

File: src/set_partner_key.c (hunk 2 above)
Comment (spalmer25) [reply src/set_partner_key.c:78]: Same as above

**Review by ajinkyasraj — APPROVED** [review]
LGTM

**Pull request merged**